feat: 허브별 배송 담당자 조회

dain391 opened the pull request:
## Issue Number
> closed #102

## 📝 Description
- 배송 담당자 조회 기능 컨트롤러 테스트 전면 구현
- 성공 케이스(2건, 0건) 테스트 추가
- 잘못된 UUID 형식 요청 시 404 응답 테스트 추가
- HUB_NOT_FOUND 예외 발생 시 404 + isSuccess=false 테스트 추가

## 🌐 Test Result
- local 환경에서 `/api/v1/deliveries-staff/{hubId}` GET 요청 테스트 완료
- MockMvc 기반 테스트 전체 통과  

## 🔎 To Reviewer
- 허브 조회 실패 케이스 처리 방식이 도메인 정책에 맞는지 확인 부탁드립니다.

dain391 committed:
merge: 충돌 해결

dain391 committed:
move: dto 구조 변경

dain391 committed:
feat: 배송 담당자 엔티티 및 타입 Enum 추가

- DeliveryStaff 엔티티 생성
- DeliveryStaffType Enum 정의

#96

dain391 committed:
feat: 배송 담당자 등록 요청 DTO 추가 및 Validation 적용

- DeliveryStaffRequest 생성
- hubId, staffType, slackId, assignOrder 등 필드 정의
- @NotNull, @Size 등 유효성 검증 추가

#96

dain391 committed:
feat: 배송 담당자 Repository 및 Service 구현

- DeliveryStaffRepository 인터페이스 생성 (JpaRepository 상속)
- DeliveryStaffService 구현
  - 허브 대기 상태에서만 배정 가능하도록 검증
  - Delivery.assignStaff 로직 호출

#96

dain391 committed:
feat: 배송 담당자 등록 컨트롤러 API 추가

- POST /api/v1/deliveries/{deliveryId}/staff
- @Valid RequestBody 적용 및 성공 시 ApiResponse 반환
  - 허브 대기 상태에서만 배정 가능하도록 검증
  - Delivery.assignStaff 로직 호출

#96

dain391 committed:
test: 배송 담당자 등록 Controller 테스트 추가

- DeliveryStaffControllerTest 작성
- 성공 및 유효성 검증 실패 케이스 테스트
- MockMvc 기반 JSON 응답 필드 검증 (isSuccess, message)
- 서비스 계층 모킹 및 request/response 직렬화 테스트 완료

#96

dain391 committed:
refactor: DeliveryRouteControllerTest 구조 수정

#96

dain391 committed:
merge: 충돌 해결

dain391 committed:
feat: 배송담당자 도메인 모델 추가

- DeliveryStaff 엔티티/모델 정의
- 기본 필드 및 생성 메서드 정리

#102

dain391 committed:
feat: 배송담당자 도메인 모델 추가

- DeliveryStaff 엔티티/모델 정의
- 기본 필드 및 생성 메서드 정리

#102

dain391 committed:
feat: 배송담당자 응답 DTO 추가

- DeliveryStaffResponse 레코드 도입
- 도메인 → 응답 변환 정리

#102

dain391 committed:
feat: 배송담당자 응답 DTO 추가

- DeliveryStaffResponse 레코드 도입
- 도메인 → 응답 변환 정리

#102

dain391 committed:
feat: 배송담당자 레포지토리/서비스 구현

- DeliveryStaffRepository 인터페이스 정의
- DeliveryStaffService: 허브별 조회 유즈케이스 추가

#102

dain391 committed:
feat: 배송담당자 레포지토리/서비스 구현

- DeliveryStaffRepository 인터페이스 정의
- DeliveryStaffService: 허브별 조회 유즈케이스 추가

#102

dain391 committed:
feat: 배송담당자 조회 컨트롤러 추가

#102

dain391 committed:
feat: 배송담당자 조회 컨트롤러 추가

#102

dain391 committed:
feat: 허브별 배송 담당자 조회 테스트 구현

- 허브 ID로 배송 담당자 조회 성공 테스트 추가 (2건)
- 조회 결과가 비어 있는 경우 테스트 추가 (0건)
- 잘못된 hubId 형식 요청 시 404 응답 검증
- HUB_NOT_FOUND 예외 발생 시 404 + isSuccess=false 응답 검증
- DeliveryStaffService mock 스텁 구성 추가

#102

dain391 committed:
feat: 허브별 배송 담당자 조회 테스트 구현

- 허브 ID로 배송 담당자 조회 성공 테스트 추가 (2건)
- 조회 결과가 비어 있는 경우 테스트 추가 (0건)
- 잘못된 hubId 형식 요청 시 404 응답 검증
- HUB_NOT_FOUND 예외 발생 시 404 + isSuccess=false 응답 검증
- DeliveryStaffService mock 스텁 구성 추가

#102

GoodNyong committed:
refactor:#76 알림 서비스 Risk Refactoring (#97)

* update: 알림 도메인 유효성 검증 메시지 메서드 추가

* refactor: 개선된 주문 알림 발송 로직 및 Gemini 연계 처리

* test: 통합 테스트를 위한 mock setting

  - OrderCreatedConsumerIT @BeforeEach Mock 설정 추가
  - DeliveryStatusChangedConsumerIT @BeforeEach Mock 설정 추가
  - Gemini, Slack Mock 응답 설정으로 실제 API 호출 방지"

* refactor: 추가된 알림 관련 오류 코드 정의

  - ErrorCode.NOTIFICATION_SEND_FAILED 추가
  - Slack 전송 실패 시 사용"

* refactor: 개선된 사용자 정보 조회 및 오류 처리 로직 추가

* test: 개선된 주문 알림 테스트 스크립트 및 데이터 구조 변경

* chore: jwt 환경 변수 설정 추가

* test: 테스트 수행 설명 추가

* refactor: 빌더 및 생성자 추가

* test: 알림 서비스 단위 테스트

sonaanweb committed:
feat: 상품 재고 증감/차감 구현 및 내부 통신용 internal 엔드포인트 추가 (#99)

* feat: 업체 내부 통신용 internal 엔드 포인트 추가

* feat: 상품 내부 통신용 internal 엔드 포인트 추가

- 재고 차감, 증가 ErrorCode 추가

* del: 사용하지 않는 코드 삭제

* del: 사용하지 않는 주석 삭제

* add: 주석 추가 및 에러코드 추가

* fix: hub-internal 엔드 포인트 추가

* add: internal swagger 태그 작성

* del: 사용하지 않는 코드 제거

Sp-PJS committed:
feat: 회원가입 승인 or 거부 및 권한설정 구현 & 로그인 시 서비스에 접근할 때 401, 500 Errror 현상 해결(추가: 마이페이지, 관리자 전용 조회 구현) (#100)

* chore: 의존성 추가 및 설정 파일 수정

* chore: 의존성 추가 및 설정 파일 수정

* chore: 에러 코드 추가

* feat: 로그인 기능 구현

* chore: Session 비활성화 코드 추가

* docs: gitignore 수정

* feat: 회원가입 기능 구현

* fix: 로그인 후 서비스 접근 시 401, 500 Error 해결 및 Header에 Id, Name, Role 전달하고 포스트맨에서 응답헤더로 볼 수 있게 수정

* chore: 코드 수정 및 1분의 시간 오차 허용

* fix: @PreAuthorize 활성화를 위한 어노테이션 추가

* feat: 회원가입 승인 or 거부 및 권한설정

* chore: Redis 캐싱을 위한 yml 파일 코드 추가

* chore: Redis 캐싱을 위한 의존성 추가

* chore: Redis 캐싱을 위한 어노테이션 추가

* chore: Redis 캐싱을 위한 캐시 매니저를 Bean으로 등록

* chore: Redis 캐싱을 위한 어노테이션 추가

* feat: 마이페이지 조회 & 관리자 전용 전체 조회(캐싱적용)

* feat: 마이페이지 & 관리자 전용 조회(캐싱 적용) 구현

dyun23 commented on delivery-service/src/main/java/com/oneforlogis/delivery/domain/model/DeliveryStaffType.java:
```diff
@@ -0,0 +1,7 @@
+package com.oneforlogis.delivery.domain.model;
+
+public enum DeliveryStaffType {
+    IN_HOUSE,   // 자사 직원
+    PARTNER,    // 제휴/외주
+    COURIER     // 택배/라이더
```
여기에 있는 타입이 Role이랑 다른걸까요??

GoodNyong committed:
feat: 알림 서비스: 배송 상태 변경 rest api (#105)

* update: 알림 도메인 유효성 검증 메시지 메서드 추가

* refactor: 개선된 주문 알림 발송 로직 및 Gemini 연계 처리

* test: 통합 테스트를 위한 mock setting

  - OrderCreatedConsumerIT @BeforeEach Mock 설정 추가
  - DeliveryStatusChangedConsumerIT @BeforeEach Mock 설정 추가
  - Gemini, Slack Mock 응답 설정으로 실제 API 호출 방지"

* refactor: 추가된 알림 관련 오류 코드 정의

  - ErrorCode.NOTIFICATION_SEND_FAILED 추가
  - Slack 전송 실패 시 사용"

* refactor: 개선된 사용자 정보 조회 및 오류 처리 로직 추가

* test: 개선된 주문 알림 테스트 스크립트 및 데이터 구조 변경

* chore: jwt 환경 변수 설정 추가

* test: 테스트 수행 설명 추가

* refactor: 빌더 및 생성자 추가

* test: 알림 서비스 단위 테스트

* feat: DeliveryStatusNotificationRequest DTO 작성

- Jakarta Validation 적용 (@NotNull, @NotBlank)
- Swagger 문서화 (@Schema)
- DeliveryStatusChangedEvent.DeliveryData와 동일한 필드 구조

* feat: NotificationService.sendDeliveryStatusNotification() 메서드 추가

- DeliveryStatusChangedConsumer 로직 재사용
- SYSTEM 타입 알림 생성
- 메시지 타입: DELIVERY_STATUS_UPDATE
- Slack 발송 실패 시 CustomException throw (HTTP 500 응답)
- DeliveryStatusChangedConsumer와 동일한 메시지 형식

* feat: 배송 상태 변경 알림 발송 API 추가

- 권한: MASTER, HUB_MANAGER, DELIVERY_MANAGER, COMPANY_MANAGER
- HTTP 201 Created 응답
- Swagger 문서화
- Jakarta Validation 자동 적용

* test: 배송 상태 변경 알림 발송 API 단위 테스트 추가

- 성공 케이스: 201 Created 응답
- 필수 필드 누락: 400 Bad Request

* test: 배송 상태 변경 알림 발송 API 단위 테스트 추가

- Test 2: 배송 상태 변경 알림 발송 추가
- 총 테스트 수: 9개 → 10개

* docs: notification/README.md 업데이트

dyun23 committed:
feat: 허브 서비스 내부용 internal 추가 및 gateway 로직 수정 및 통합 swagger 추가 (#103)

* chore: /api/v1/internal 경로 권한 없이 접근 허용

* refactor: 최단 거리 db 저장 제거

* feat: Hub에 Internal 엔드포인트 추가(#98)

* design: 스웨거 표시 이름 변경

* feat: docker-compose 모든 서비스 추가

* chore: swagger 관련 설정 추가 및 환경변수 분리

* chore: openapi 패키지 추가

* refactor: webflux 변환으로 코드 일부 수정

* chore: swagger config 서버 추가 및 헤더 jwt 토큰으로 변환

* chore: swagger 관련 경로 허용 추가

* refactor: 화이트리스트 filter로 이전

* chore: cors config 추가

dyun23 committed:
fix: 충돌 해결 부분 수정 (#107)

dain391 committed:
Merge remote-tracking branch 'origin/feature/#102-read-delivery-staff-by-hub' into feature/#102-read-delivery-staff-by-hub

# Conflicts:
#	delivery-service/src/test/java/com/oneforlogis/delivery/presentation/controller/DeliveryControllerTest.java

dyun23 reviewed (dismissed):
수고하셨습니당
